test/Interop/C/bounds-safety/Inputs/bounds-attributed-function.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <ptrcheck.h>
+#include <stdarg.h>
+
+#ifndef __unsafe_late_const
+#define __unsafe_late_const
+#endif
+
+int * __counted_by(len) a(int *__counted_by(len) p, int len);
+char * __counted_by(len) b(char *__counted_by(len) p, int len);
+char * __sized_by(len) c(char *__sized_by(len) p, int len);
+void * __sized_by(len) d(void *__sized_by(len) p, int len);
+int * __sized_by(len) e(int *__sized_by(len) p, int len);
+
+int * __ended_by(end) f(int *__ended_by(end) p, int * end);
+void * __ended_by(end) g(void *__ended_by(end) p, void * end);
+
+char * __null_terminated h(char *__null_terminated p);
+const char * i(const char * p);
+
+char * __single j(char *__single p);
+void *__single k(void *__single p);
+
+#if __has_ptrcheck
```

Comment:
Why are only `__indexable` and `__bidi_indexable` guarded with `__has_ptrcheck`?

Reply:
Other macros defined in `ptrcheck.h` are defined to nothing when `-fbounds-safety` is not enabled, these two are intentionally left undefined since they can affect ABI, so you don't want one TU thinking they are normal pointers and one TU expanding them to wide pointers